File: Backend/src/main/java/com/colocation/services/ContactService.java
package com.colocation.services;

import com.colocation.models.ContactModel;

import java.util.List;
import java.util.Optional;

public interface ContactService {

    ContactModel createContact(ContactModel contact);

    Optional<ContactModel> getContactById(Long id);
    List<ContactModel> getAllContacts();
    void deleteContact(Long id);
}

File: Frontend/src/app/Service/Contact/contact.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Contact } from 'src/app/models/ContactModel';
@Injectable({
  providedIn: 'root'
})
export class ContactService {
  private apiUrl = 'http://localhost:8089/api/contacts';

  constructor(private http: HttpClient) { }

  createContact(contactData: Contact): Observable<Contact> {
    return this.http.post<Contact>(`${this.apiUrl}/add`, contactData);
  }

  getContactById(id: number): Observable<Contact> {
    return this.http.get<Contact>(`${this.apiUrl}/${id}`);
  }

  getAllContacts(): Observable<Contact[]> {
    return this.http.get<Contact[]>(this.apiUrl);
  }

  deleteContact(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }
}
